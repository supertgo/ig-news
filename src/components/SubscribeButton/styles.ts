import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    width: 26rem;
    height: 6.4rem;
    border: 0;
    border-radius: ${theme.spacings.medium};
    background: ${theme.colors.yellow};
    color: ${theme.colors.background};
    font-size: ${theme.font.sizes.xlarge};
    font-weight: ${theme.font.bold};

    display: flex;
    align-items: center;
    justify-content: center;
    transition: filter 0.2s;

    &:hover {
      filter: brightness(0.8);
    }
  `}
`;
