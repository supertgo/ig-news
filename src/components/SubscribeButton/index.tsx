import * as S from './styles';

const SubscribeButton = () => <S.Wrapper>Subscribe Now</S.Wrapper>;

export default SubscribeButton;
