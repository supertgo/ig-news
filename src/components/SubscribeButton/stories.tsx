import { Story, Meta } from '@storybook/react';
import SubscribeButton from '.';

export default {
  title: 'SubscribeButton',
  component: SubscribeButton
} as Meta;

export const Default: Story = () => <SubscribeButton />;
