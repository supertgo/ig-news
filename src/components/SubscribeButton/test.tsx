import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import SubscribeButton from '.';

describe('<SubscribeButton />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<SubscribeButton />);

    expect(
      screen.getByRole('button', { name: /Subscribe Now/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
