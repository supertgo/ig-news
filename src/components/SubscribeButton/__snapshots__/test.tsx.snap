// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscribeButton /> should render the heading 1`] = `
.c0 {
  width: 26rem;
  height: 6.4rem;
  border: 0;
  border-radius: 3.2rem;
  background: #EBA417;
  color: #121414;
  font-size: 2.0rem;
  font-weight: 600;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: filter 0.2s;
  transition: filter 0.2s;
}

.c0:hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
}

<button
  class="c0"
>
  Subscribe Now
</button>
`;
